### PR TITLE
bug fixes in watcherfactory shouldnotify

### PR DESCRIPTION
**Pull request by kylezeeuwen**

### CODE REVIEW

File: src/factories/watchableListFactory.js [hunk 1]
```diff
@@ -21,7 +21,7 @@ class WatchableList {
```
Comment (kylezeeuwen) [inline src/factories/watchableListFactory.js]: no functional change just remove the `return`s added by coffee -> JS process

File: test/karma/factories/watcherFactory.test.js [hunk 2]
```diff
@@ -4,13 +4,79 @@ describe('WatcherFactory', function() {
...
 
   describe('shouldNotify', function() {
-    it('returns true if the data being watched has changed', inject(function (WatcherFactory) {
-      const handler = function () {};
-      const watchPath = 'a';
-      const watcher = WatcherFactory.create(watchPath, handler);
 
-      expect(watcher.shouldNotify('a', 'b')).toBe(true);
-    }));
+    // NB Each entry in the array is a test, each test can be a
```
Comment (lee-borsboom) [inline test/karma/factories/watcherFactory.test.js]: You've labelled both options as A

File: test/karma/factories/watcherFactory.test.js [hunk 3]
```diff
@@ -4,13 +4,79 @@ describe('WatcherFactory', function() {
...
+
+    function normalizeTestDefinition(testDefinition) {
+      if (_.isString(testDefinition)) {
+        return interpretTestDsl(testDefinition);
+      } else {
+        return testDefinition;
+      }
+    }
+
+    function interpretTestDsl(inputString) {
```
Comment (lee-borsboom) [inline test/karma/factories/watcherFactory.test.js]: grammar

File: test/karma/factories/watcherFactory.test.js (hunk 2 above)
Comment (kylezeeuwen) [inline test/karma/factories/watcherFactory.test.js]: done

File: test/karma/factories/watcherFactory.test.js (hunk 3 above)
Comment (kylezeeuwen) [inline test/karma/factories/watcherFactory.test.js]: done

**Pull request merged**